drivers/soundwire/intel_init.c

```diff
@@ -56,26 +92,24 @@ static bool is_link_enabled(struct fwnode_handle *fw_node, int i)
 
 static int sdw_intel_cleanup(struct sdw_intel_ctx *ctx)
 {
-	struct sdw_intel_link_res *link = ctx->links;
+	struct sdw_intel_link_dev *ldev = ctx->ldev;
 	u32 link_mask;
 	int i;
 
-	if (!link)
+	if (!ldev)
 		return 0;
```

Comment:
unrelated to this patch: this function always returns 0 and its return code is never checked. Time to make it void?

Reply:
indeed, I don't know why it uses an 'int', maybe legacy code from the past. 
it would make sense to change this, I am busy on other things so let's see who beats me to it, @lyakh or @bardliao?